ttnn/cpp/ttnn/operations/moreh/moreh_dot_op/device/moreh_dot_device_operation.hpp

```diff
@@ -0,0 +1,70 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+
+#include <optional>
+#include <variant>
+
+#include "ttnn/tensor/tensor.hpp"
+#include "ttnn/device_operation.hpp"
+#include "ttnn/types.hpp"
+#include "ttnn/decorators.hpp"
+
+
+namespace ttnn::operations::moreh::moreh_dot_op {
+
+struct MorehDotOperation {
+    struct operation_attributes_t {
+        const DataType output_dtype;
+        const std::optional<MemoryConfig> output_mem_config;
```

Comment:
```suggestion
        const MemoryConfig output_memory_config;
```

https://github.com/tenstorrent/tt-metal/blob/dd62884e2ea0bc4dd8bbd072e0becab647298658/ttnn/cpp/ttnn/operations/eltwise/binary/device/binary_device_operation.hpp#L35
Looking at this code, it seems like optional may not be needed. Is it possible to remove the optional?

Reply:
@o2buzzle

Reply:
https://github.com/tenstorrent/tt-metal/pull/12095#discussion_r1738548274

There was this review on my (older) PR that requested me to make this change, hence we're basing our code on that.

Reply:
Sorry, I think I made a mistake in my comment. Inside `operation_attributes_t`, it’s not optional, but when receiving it from Python, for example, in `invoke`, it is optional